### ISSUE TITLE
Attenuated high frequencies on DMG

### ISSUE THREAD

**Issue by jkotlinski** [LIJI32/SameBoy#348]
When comparing an ad hoc white noise signal played back through wave channel, I found Sameboy 0.13.6 remarkably close to a real DMG A/B/C.
While the digital signal generation seems impeccable, I found a difference in high-end of the spectrum, giving real hardware an edge due to a sharper, crisper and ultimately more satisfying sound.
The difference is confirmed by spectral analysis, which shows that Sameboy signal is attenuated by at least 1 dB for frequencies over ~13 kHz.

Test files attached.

### Real DMG
<img width="571" alt="realdmg" src="https://user-images.githubusercontent.com/3259963/109192807-fe96e100-7797-11eb-8c79-cf2753ff7348.png">

### Sameboy 0.13.6
<img width="572" alt="sameboy" src="https://user-images.githubusercontent.com/3259963/109192809-fe96e100-7797-11eb-9ecf-7406ea0fe8de.png">
[noise.zip](https://github.com/LIJI32/SameBoy/files/6044784/noise.zip)

**Comment by jkotlinski** [comment]
After inspecting the code, it feels likely that this error is introduced when downsampling APU output to 48 kHz.

**Comment by jkotlinski** [comment]
As a sanity check, I made the same comparison with Gambatte (configured to use Highest Quality FIR downsampler).
Here, there is no serious roll-off until 17 kHz. I think the difference is audible and Gambatte certainly seems closer to the real thing.

<img width="911" alt="gambatteoverlay" src="https://user-images.githubusercontent.com/3259963/109638395-4332cb80-7b4e-11eb-89a9-a00d6b4a95fb.png">

**Comment by LIJI32** [comment]
Could also be different high-pass-filter constants as well as the lack of a low pass filter in SameBoy.

**Comment by jkotlinski** [comment]
Maybe [Blip Buffer](https://github.com/eriser/blip-buffer) could be incorporated for the resampling?

IMO this would be the last hurdle for 100% audio emulation.

**Comment by jkotlinski** [comment]
I investigated further. In the samples attached in this message, it looks like most of the difference is from wave refresh audio spikes being more pronounced on real DMG.
So in this particular case, Sameboy is "too good" :)  which is maybe not a disaster.

[808.zip](https://github.com/LIJI32/SameBoy/files/6315608/808.zip)

**Comment by jkotlinski** [comment]
As a summary, my analysis now is that the DC blocker emulation is a bit off. Sometimes it sounds better, sometimes it sounds worse.

You can choose for yourself, if it is something worth fixing :-)

**Comment by lpla** [comment]
Definitely, the 808.zip comparison is clearly audible. How other emulators behave with this test?

**Comment by jkotlinski** [comment]
Gambatte is very close to Sameboy.
BGB is different, also not so much like real hardware.
